internal/app/focus_manager.go

```diff
@@ -73,6 +85,10 @@ func (f *FocusManager) focus(obj fyne.Focusable) {
 		obj = nil
 	}
 
+	if hid, ok := obj.(fyne.CanvasObject); ok && !hid.Visible() {
```

Comment:
Is this catching a specific case or a general concern that we were focusing hidden objects?
I ask because `Visible()` is not returning "is currently visible" but more like "has not been asked to hide"...
i.e. a tab container will hide tabs that are not current - but the contents of that tab will still return `true` for `Visible()`...

Reply:
This was only a general concern. You’re right about the container stuff and I did not think of it.
But after doing so, I’d say that implementing a walk, to see if this object might be hidden because a parent is, is too much effort just to calm a general concern.

Reply:
Would it be worth using `driver.WalkVisibleObjectTree` instead of `driver.WalkCompleteObjectTree`?
That is pretty much what it is for.

Reply:
Hell, yes! Why I didn’t see this?